backend/src/main/java/com/festago/school/repository/v1/SchoolV1QueryDslRepository.java

```diff
@@ -0,0 +1,132 @@
+                                socialMedia.mediaType,
+                                socialMedia.name,
+                                socialMedia.logoUrl,
+                                socialMedia.url
+                            ).skipNulls()
+                        )
+                    )
+                )
+            );
+
+        if (response.isEmpty()) {
+            return Optional.empty();
+        }
+
+        return Optional.of(response.get(0));
+    }
+
+    public SliceResponse<SchoolFestivalV1Response> findFestivalsBySchoolId(Long schoolId, LocalDate today,
+                                                                           SchoolFestivalV1SearchCondition searchCondition) {
+        List<SchoolFestivalV1Response> result =
+            select(new QSchoolFestivalV1Response(festival.id,
+                    festival.name,
+                    festival.startDate,
+                    festival.endDate,
+                    festival.thumbnail,
+                    festivalQueryInfo.artistInfo
+                )
+            )
+                .from(festival)
+                .where(festival.school.id.eq(schoolId),
+                    addPhaseOption(searchCondition.isPast(), today),
+                    addPagingOption(searchCondition.lastFestivalId(), searchCondition.lastStartDate(),
+                        searchCondition.isPast()))
+                .leftJoin(festivalQueryInfo).on(festivalQueryInfo.festivalId.eq(festival.id))
+                .orderBy(addOrderOption(searchCondition.isPast()))
+                .limit(searchCondition.size() + NEXT_PAGE_CHECK_NUMBER)
+                .fetch();
+
+        return createResponse(searchCondition.size(), result);
+    }
+
+    private BooleanExpression addPhaseOption(boolean isPast, LocalDate today) {
+        if (isPast) {
+            return festival.endDate.lt(today);
+        }
+
+        return festival.endDate.goe(today);
+    }
+
+    private BooleanExpression addPagingOption(Long lastFestivalId, LocalDate lastStartDate, boolean isPast) {
+        if (isNotFirstPage(lastFestivalId, lastStartDate)) {
+            if (isPast) {
+                return festival.startDate.lt(lastStartDate)
+                    .or(festival.startDate.eq(lastStartDate)
+                        .and(festival.id.gt(lastFestivalId)));
+            }
+            return festival.startDate.gt(lastStartDate)
+                .or(festival.startDate.eq(lastStartDate)
+                    .and(festival.id.gt(lastFestivalId)));
+        }
+
+        return null;
+    }
+
+    private boolean isNotFirstPage(Long lastFestivalId, LocalDate lastStartDate) {
+        return lastFestivalId != null && lastStartDate != null;
+    }
+
+    private OrderSpecifier<LocalDate>[] addOrderOption(boolean isPast) {
+        if (isPast) {
+            return new OrderSpecifier[]{festival.endDate.desc()};
+        }
+
+        return new OrderSpecifier[]{festival.startDate.asc()};
+    }
```

Comment:
단 하나의 OrderSpecifier를 사용할거면 배열을 사용할 필요는 없어 보이네요!
```suggestion
    private OrderSpecifier<LocalDate> addOrderOption(boolean isPast) {
        if (isPast) {
            return festival.endDate.desc()
        }

        return festival.startDate.asc();
    }
```